sdk/src/connections/offline/offline_depth_sensor.cpp

```diff
@@ -1,25 +1,66 @@
 #include "offline_depth_sensor.h"
+#include "tofi/tofi_config.h"
 #include <cstring>
+#include <dirent.h>
 #include <fstream>
+#include <glog/logging.h>
 #include <iostream>
+#include <iterator>
+#include <map>
+#include <vector>
 
 OfflineDepthSensor::OfflineDepthSensor(std::string path) : m_path(path) {
     m_sensorDetails.connectionType = aditof::ConnectionType::OFFLINE;
+    m_connectionType = "offline";
 }
 
-OfflineDepthSensor::~OfflineDepthSensor() = default;
+OfflineDepthSensor::~OfflineDepthSensor() {
+    for (auto index : m_frameTypes) {
+        delete[] index.second.first;
+    }
+    m_frameTypes.clear();
+}
 
 aditof::Status OfflineDepthSensor::getFrame(uint16_t *buffer) {
-    std::ifstream ifs(m_path.c_str());
-    std::string content((std::istreambuf_iterator<char>(ifs)),
-                        (std::istreambuf_iterator<char>()));
-    for (int i = 0; i < content.size(); i += 2) {
-        buffer[i / 2] = *(uint16_t *)&content[i];
+    auto frame = m_frameTypes.find(m_frameTypeSelected + ".bin");
+    if (frame == m_frameTypes.end()) {
+        LOG(WARNING) << "get frame error " << m_frameTypeSelected
+                     << " is not found";
+        return aditof::Status::GENERIC_ERROR;
     }
+    memcpy(buffer, frame->second.first, frame->second.second);
     return aditof::Status::OK;
 }
 
-aditof::Status OfflineDepthSensor::open() { return aditof::Status::OK; }
+aditof::Status OfflineDepthSensor::open() {
+    std::vector<std::string> frameTypesResources;
+    std::streampos size;
+    std::string fileName;
+    uint16_t *buffer;
+    struct dirent *entry;
+    DIR *dir = opendir(m_path.c_str());
+    while ((entry = readdir(dir)) != NULL) {
+        fileName = entry->d_name;
+        if (fileName != "." && fileName != "..")
+            frameTypesResources.push_back(fileName);
+    }
+
+    for (int index = 0; index < frameTypesResources.size(); index++) {
+        std::ifstream ifs;
+        ifs.open(m_path + '/' + frameTypesResources[index],
+                 std::ios::in | std::ios::binary);
+        std::string content((std::istreambuf_iterator<char>(ifs)),
+                            (std::istreambuf_iterator<char>()));
+        buffer = new uint16_t[content.size()];
+        for (int i = 0; i < content.size(); i += 2) {
```

Comment:
You can replace this for with a simple memcpy.

Reply:
For replaced with memcpy.